tests-arquillian/src/test/java/org/jboss/weld/tests/interceptors/bridgemethods/hierarchy/InterceptorBridgeMethodTest.java

```diff
@@ -10,43 +26,150 @@
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
 import org.jboss.weld.test.util.ActionSequence;
 import org.jboss.weld.test.util.Utils;
+import org.junit.Ignore;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
 /**
  *
  * @author Martin Kouba
  * @see WELD-1672
+ * @see WELD-2414
  */
 @RunWith(Arquillian.class)
 public class InterceptorBridgeMethodTest {
 
     @Deployment
     public static JavaArchive createTestArchive() {
-        return ShrinkWrap.create(BeanArchive.class, Utils.getDeploymentNameAsHash(InterceptorBridgeMethodTest.class)).intercept(MissileInterceptor.class)
-                .addPackage(InterceptorBridgeMethodTest.class.getPackage()).addClass(ActionSequence.class);
+        return ShrinkWrap
+                .create(BeanArchive.class,
+                        Utils.getDeploymentNameAsHash(InterceptorBridgeMethodTest.class))
+                .intercept(MissileInterceptor.class)
+                .addPackage(InterceptorBridgeMethodTest.class.getPackage())
+                .addClass(ActionSequence.class);
     }
 
     @Test
-    public void testInterception(Child child) {
+    public void testChild(@Juicy Child child) {
+        // Child extends Parent<String> implements Base<T>
+        reset();
+        child.invokeA("foo");
+        verify(Child.class);
         reset();
-        child.invoke("foo");
-        verify();
+        child.getA();
+        verify(Parent.class);
+        reset();
+        child.invokeB("foo");
+        verify(Parent.class);
+        reset();
+        child.getB();
+        verify(Parent.class);
+        reset();
+        child.invokeDefault("foo");
+        verify(Parent.class);
+
         reset();
         Parent<String> parent = child;
-        parent.invoke("foo");
-        verify();
+        parent.invokeA("foo");
+        verify(Child.class);
+    }
+
+    @Test
+    public void testJuicyBase(@Juicy Base<String> base) {
+        // Child gets injected
+        reset();
+        base.invokeA("foo");
+        verify(Child.class);
+        reset();
+        base.getA();
+        verify(Parent.class);
+        reset();
+        base.invokeB("foo");
+        verify(Parent.class);
+        reset();
+        base.getB();
+        verify(Parent.class);
+        reset();
+        base.invokeDefault("foo");
+        verify(Parent.class);
+
+    }
+
+    @Test
+    public void testSpecialBase(SpecialBase special) {
+        // SpecialChild gets injected
+        reset();
+        special.getA();
+        verify(SpecialChild.class);
+        reset();
+        special.invokeB("foo");
+        verify(SpecialChild.class);
+        reset();
+        special.getB();
+        verify(SpecialParent.class);
+        reset();
+        special.invokeDefault("foo");
+        verify(Base.class);
+    }
+
+    @Test
+    @Ignore("WELD-2424")
```

Comment:
Just a side note - once we merge this, we should link this test to WELD-2424 issue so that it's clear how the use case looks like. Don't know about you, but I sure won't remember there was a test added back here :)

Reply:
Yes, of course. I will add a link to this line once the PR is merged.